core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/CommandLineArgumentHandler.java

```diff
@@ -0,0 +1,240 @@
+				this.flagsValid = false;
+				return;
+			}
+
+			if (f.hasParameter()) {
+				flagExpectingPara = f;
+			} else {
+				this.parameters.put(f, "");
+			}
+
+			switch (f) {
+			case HELP:
+				printHelp();
+				return;
+			case VERSION:
+				printVersion();
+				return;
+			case NOTICE:
+				printNotice();
+				return;
+			default:
+				// DO nothing;
+				break;
+			}
+		}
+		output("This is Amy. Copyright (c) 2018 the Amy project authors. For help run with flag -h.");
+	}
+
+	/**
+	 * @return whether the program should continue with normal execution considering these command line flags
+	 */
+	public boolean shouldProgramContinue() {
+		if (!this.flagsValid)
+			return false;
+		for (Flag f : this.parameters.keySet()) {
+			if (f.isStopExecutaion())
+				return false;
+		}
+		return true;
+	}
+
+	private void printHelp() {
+		output("This is Amy. A open source personal assitance system.");
+		output("Developed by a group of students at University of Stuttgart.");
+		output("This is a research project. No functionality is tested. There may be harmful errors.");
+		output("");
+		output("Copyright (c) 2018 the Amy project authors.");
+		output("");
+		output("For information about the license please start the program with --notice");
+		output("");
+		output("For further information about the project see: github.com/AmyAssist");
+		output("");
+		output("For help with the command line interface, after the program has started please enter '?'.");
+		output("");
+		output("Command line flags:");
+		output("");
+		for (Flag f : Flag.values()) {
+			String s = "  ";
+			if (f.getShortVariant() != "") {
+				s += f.getShortVariant();
+			}
+			if (f.getShortVariant() != "" && f.getLongVariant() != "") {
+				s += " or ";
+			}
+			if (f.getLongVariant() != "") {
+				s += f.getLongVariant();
+			}
+
+			s += " : " + f.getDescription();
+
+			output(s);
+		}
+	}
+
+	private void printVersion() {
+		output("Amy core version: " + getClass().getPackage().getImplementationVersion());
+	}
+
+	private void printNotice() {
+		output("Copyright (c) 2018 the Amy project authors.\n" + " \n" + "  SPDX-License-Identifier: Apache-2.0\n"
+				+ " \n" + "  Licensed under the Apache License, Version 2.0 (the \"License\");\n"
+				+ "  you may not use this file except in compliance with the License.\n"
+				+ "  You may obtain a copy of the License at\n" + " \n"
+				+ "    http://www.apache.org/licenses/LICENSE-2.0\n" + " \n"
+				+ "  Unless required by applicable law or agreed to in writing, software\n"
+				+ "  distributed under the License is distributed on an \"AS IS\" BASIS,\n"
+				+ "  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n"
+				+ "  See the License for the specific language governing permissions and\n"
+				+ "  limitations under the License.\n" + " \n" + "  For more information see notice.md");
+	}
+
+	private void output(String s) {
+		System.out.println(s);
+	}
+
+	private enum Flag {
```

Comment:
maybe extract the enum in extra file and move all in a package named commandlinearguments